front_end/webserver.py

```diff
@@ -231,67 +252,85 @@ class AssignmentHandler(BaseUserHandler):
     def get(self, course, assignment):
         try:
             show = show_hidden(self)
-            self.render("assignment.html", courses=get_courses(show), assignments=get_assignments(course, show), problems=get_problems(course, assignment, show), course_basics=get_course_basics(course), assignment_basics=get_assignment_basics(course, assignment), assignment_details=get_assignment_details(course, assignment, True), user_id=user_id_var.get(), user_logged_in=user_logged_in_var.get())
+            self.render("assignment.html", courses=content.get_courses(show), assignments=content.get_assignments(course, show), problems=content.get_problems(course, assignment, show), course_basics=content.get_course_basics(course), assignment_basics=content.get_assignment_basics(course, assignment), assignment_details=content.get_assignment_details(course, assignment, True), user_id=user_id_var.get(), user_logged_in=user_logged_in_var.get())
         except Exception as inst:
             render_error(self, traceback.format_exc())
 
 class EditAssignmentHandler(BaseUserHandler):
     def get(self, course, assignment):
         try:
-            self.render("edit_assignment.html", courses=get_courses(), assignments=get_assignments(course), problems=get_problems(course, assignment), course_basics=get_course_basics(course), assignment_basics=get_assignment_basics(course, assignment), assignment_details=get_assignment_details(course, assignment), result=None, user_id=user_id_var.get(), user_logged_in=user_logged_in_var.get())
+            self.render("edit_assignment.html", courses=content.get_courses(), assignments=content.get_assignments(course), problems=content.get_problems(course, assignment), course_basics=content.get_course_basics(course), assignment_basics=content.get_assignment_basics(course, assignment), assignment_details=content.get_assignment_details(course, assignment), result=None, user_id=user_id_var.get(), user_logged_in=user_logged_in_var.get())
         except Exception as inst:
             render_error(self, traceback.format_exc())
 
     def post(self, course, assignment):
         try:
-            assignment_basics = get_assignment_basics(course, assignment)
-            assignment_basics["title"] = self.get_body_argument("title").strip()
-            assignment_basics["visible"] = self.get_body_argument("is_visible") == "Yes"
-            assignment_details = {"introduction": self.get_body_argument("introduction").strip()}
+            new_assignment = True
+            if content.check_assignment_exists(assignment):
+                new_assignment = False
+                assignment_details = content.get_assignment_details(course, assignment)
+            assignment_basics = content.get_assignment_basics(course, assignment)
+
+            if new_assignment:
+                assignment_basics["title"] = self.get_body_argument("title").strip()
```

Comment:
Please create local variables for title and introduction and visible so you don't need to repeatedly pull the body arguments.